dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -227,10 +229,14 @@ public T deserialize(Response response) {
                             "Response is missing Content-Type header",
                             SafeArg.of("received", response.headers().keySet()));
                 }
-                EncodingDeserializerContainer<T> container = getResponseDeserializer(contentType.get());
-                return container.deserializer.deserialize(response.body());
+                Encoding.Deserializer<T> deserializer = getResponseDeserializer(contentType.get());
+                // deserializer is responsible for closing the response
+                closeResponse = false;
```

Comment:
nit: I'd actually prefer to set this after executing the deserializer and before returning it's result, that way if it throws and fails in an unexpected way we don't risk leaking resources.
That may result in multiple close invocations, but it beats the alternative of leaking resources in the case where a deserializer fails prior to a try-with-resources block.

Reply:
Have also added a test for this